batch7crud-roll1/src/main/java/com/lftechnology/batch7crud/controller/EmployeeController.java

```diff
@@ -0,0 +1,216 @@
+
+import java.io.IOException;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Servlet implementation class EmployeeController
+ */
+@WebServlet({ "/employees/*" })
+public class EmployeeController extends HttpServlet {
+
+	private static final long serialVersionUID = 1L;
+	EmployeeService employeeService = new EmployeeService();
+	private static Logger logger = Logger.getLogger(EmployeeController.class.getName());
+	private static final String CREATE = "create";
+	private static final String DELETE = "delete";
+	private static final String EDIT = "edit";
+
+
+	/**
+	 * @see HttpServlet#HttpServlet()
+	 */
+	public EmployeeController() {
+
+		super();
+	}
+
+	/**
+	 * @see HttpServlet#doGet(HttpServletRequest request, HttpServletResponse response)
+	 */
+	@Override
+	protected void doGet(
+					HttpServletRequest request, HttpServletResponse response)
+					throws ServletException, IOException {
+
+		String requestInfo = request.getPathInfo();
+
+		if (requestInfo == null) {
+			list(request, response, 1);
+		}
+		else {
+
+			String[] pathParts = requestInfo.split("/");
+
+			if (CREATE.equals(pathParts[1])) {
+				create(request, response);
+			}
+			else if (EDIT.equals(pathParts[2])) {
+				int id = TypeCaster.toInt(pathParts[1]);
+				edit(request, response, id);
+			}
+		}
+	}
+
+	/**
+	 * @see HttpServlet#doPost(HttpServletRequest request, HttpServletResponse response)
+	 */
+	@Override
+	protected void doPost(
+					HttpServletRequest request, HttpServletResponse response)
+					throws ServletException, IOException {
+
+		String requestInfo = request.getPathInfo();
+
+		String[] pathParts = requestInfo.split("/");
+
+		if (CREATE.equals(pathParts[1])) {
+			createProcess(request, response);
+		}
+		else if (EDIT.equals(pathParts[2])) {
+			int id = TypeCaster.toInt(pathParts[1]);
+			editProcess(request, response, id);
+		}
+		else if (DELETE.equals(pathParts[2])) {
+			int id = TypeCaster.toInt(pathParts[1]);
+			deleteProcess(request, response, id);
+		}
+	}
+
+	private void list(
+					HttpServletRequest request, HttpServletResponse response,
+					int page)
+					throws ServletException, IOException {
+
+		try {
+			List<Employee> employees = employeeService.fetch();
+			request.setAttribute("employees", employees);
+			request.getServletContext().getRequestDispatcher(
+							"/WEB-INF/views/employee/list.jsp").forward(
+							request, response);
+		}
+		catch (DataException e) {
+
+			logger.log(Level.SEVERE, e.getMessage());
+			response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+			request.setAttribute("message", e.getMessage());
+			request.getServletContext().getRequestDispatcher(
+							"/WEB-INF/views/error.jsp").forward(
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "/WEB-INF/views/error.jsp" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)